src/components/generic/forms/input/Input.tsx

```diff
@@ -22,6 +23,7 @@ export function Input<T>(props: InputProps<T>) {
       helperText={fieldState.error?.message ?? ' '}
       inputProps={{ 'aria-label': props.label }}
       {...otherFieldProps}
+      onBlur={props.onBlur}
       inputRef={ref}
     />
   );
```

Comment:
This is the one thing I would change in our practices if I had the chance. Altough I understand the reasoning, this creates components that are too stiff to special scenarios. Alternatively we should make these components appropriately generic so that they expose all the APIs that one may need to use, which is a bit of a YAGNI. Altough more verbose, I would prefer to do the setup of inputs on a case by case basis using the HOC.

Reply:
For this kind of component, I would accept the `TextFieldProps` and spread all the props on the `TextField`.